hooks/pre-push.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -eu
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+mvn checkstyle:checkstyle --fail-at-end
```

Comment:
If this runs checkstyle on everything (versus just the modified files) it'll probably take 10s of seconds to run

Reply:
I saw the script at https://github.com/hdpe/maven-checkstyle-pre-commit/blob/master/pre-commit but the licensing is not clear so decided not to add it. It takes about 30s to runs checkstyle on everything and this is only done pre push (so should be less often then if we do pre commit). I think 30s is not too bad considering that the CI thing takes 1-2 hours and we can save resource on CI

Reply:
Also the hook is optional. If you don't run the setup script then it wont be applied.